amy/emails/models.py

```diff
@@ -111,12 +111,44 @@ def get_absolute_url(self) -> str:
 
 
 class ScheduledEmailStatus(models.TextChoices):
-    SCHEDULED = "scheduled"
-    LOCKED = "locked"
-    RUNNING = "running"
-    SUCCEEDED = "succeeded"
-    FAILED = "failed"
-    CANCELLED = "cancelled"
+    SCHEDULED = "scheduled"  # editing, cancelling allowed
+    LOCKED = "locked"  # nothing allowed
+    RUNNING = "running"  # nothing allowed
+    SUCCEEDED = "succeeded"  # editing allowed with note about re-sending
+    FAILED = "failed"  # editing, cancelling allowed
+    CANCELLED = "cancelled"  # allowed to re-schedule
+
+
+# List of states when specific actions are allowed. This is used in per-object
+# permissions to block or allow specific actions.
+ScheduledEmailStatusActions = {
+    "edit": [
+        ScheduledEmailStatus.SCHEDULED,
+        ScheduledEmailStatus.FAILED,
+        # note: will not re-send the email or affect emails already sent
+        ScheduledEmailStatus.SUCCEEDED,
```

Comment:
I admit it's debatable. I wanted to not prevent user from amending the email just because I may not see the use case they may have.

@maneesha Once email is sent successfully, should users be able to edit it? This will not affect the email whatsoever, including re-sending it.

Reply:
@maneesha decided to block editing.

Reply:
We should actually have enforcement that an email **can not** be edited after it has been sent, as this would create a situation where what is in AMY is different than what was actually set.